pkg/services/ngalert/api/generated_base_api_provisioning.go

```diff
@@ -46,7 +46,9 @@ type ProvisioningApi interface {
 func (f *ProvisioningApiHandler) RouteDeleteAlertRule(ctx *models.ReqContext) response.Response {
 	// Parse Path Parameters
 	uIDParam := web.Params(ctx.Req)[":UID"]
-	return f.handleRouteDeleteAlertRule(ctx, uIDParam)
+	// Get Header
+	disablingProvenanceHeader := ctx.Req.Header.Get("x-disable-provenance")
+	return f.handleRouteDeleteAlertRule(ctx, uIDParam, disablingProvenanceHeader == "true")
```

Comment:
Should we check that the header contains a valid boolean?

Reply:
I thought about it, but are checks necessary if we are just looking for the word `"true"`? We don't really care about anything else. Everything else would be `false`

I like this way better as it implies less code and checks. What are your thoughts about it?

Reply:
A thought I had is what if someone puts `ttrue` and doesn't notice until they've provisioned all their rules?

Reply:
A thought I had is what if someone puts `ttrue` and doesn't notice until after they've provisioned all their rules?

Reply:
Fair point

Reply:
I agree with George, we shouldn't check the content of the header. Checking if it exists should be enough, I wouldn't expect a value, as it's redundant.